### PR TITLE
DAOS-6000: fix memory leak in vos key_punch for akey case

### PR DESCRIPTION
**Pull request by marcelarosalesj**
Signed-off-by: Marcela Rosales <marcela.a.rosales.jimenez@intel.com>

### CODE REVIEW

File: src/vos/vos_obj.c [hunk 1]
```diff
@@ -163,7 +163,7 @@ key_punch(struct vos_object *obj, daos_epoch_t epoch, uint32_t pm_ver,
 	struct vos_ilog_info	 akey_info = {0};
 	daos_epoch_range_t	 epr = {0, epoch};
 	d_iov_t			 riov;
-	daos_handle_t		 toh;
+	daos_handle_t		 toh = DAOS_HDL_INVAL ;
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/vos/vos_obj.c:166]: this one should be fixed.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
just fix the checkpatch issue, otherwise, looks good to me

**Review by NiuYawei — APPROVED** [review]

**Review by Nasf-Fan — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by ashleypittman** [comment]
I just want to add that it's great to see issues like that that we're working through the memcheck errors and using it to fix real issues.

**Review by ryon-jensen — APPROVED** [review]

**Comment by jolivier23** [comment]
> I just want to add that it's great to see issues like that that we're working through the memcheck errors and using it to fix real issues.

yeah, this is the first real issue we've fixed so far.   I'm happy to have memcheck running and also to get rid of some of the suppressions that may mask any new issues.

**Pull request merged**